bolt-contracts/src/contracts/BoltManager.sol

```diff
@@ -66,21 +119,154 @@ contract BoltManager is IBoltManager, Ownable {
+    /// @return status The status of the proposer, including their operator and active stake.
+    function getProposerStatus(
+        bytes32 pubkeyHash
+    ) public view returns (IBoltValidators.ProposerStatus memory status) {
+        if (pubkeyHash == bytes32(0)) {
+            revert InvalidQuery();
+        }
+
+        uint48 epochStartTs = getEpochStartTs(getEpochAtTs(Time.timestamp()));
+        IBoltValidators.Validator memory validator = validators.getValidatorByPubkeyHash(pubkeyHash);
+
+        Operator memory operator = operators.get(validator.authorizedOperator);
+
+        status.pubkeyHash = pubkeyHash;
+        status.active = validator.exists;
+        status.operator = validator.authorizedOperator;
+        status.operatorRPC = operator.rpc;
+
+        (uint48 enabledTime, uint48 disabledTime) = operators.getTimes(validator.authorizedOperator);
+        if (!_wasEnabledAt(enabledTime, disabledTime, epochStartTs)) {
+            return status;
+        }
+
+        (status.collaterals, status.amounts) =
+            IBoltMiddleware(operator.middleware).getOperatorCollaterals(validator.authorizedOperator);
+
+        return status;
+    }
+
+    /// @notice Get the total amount staked of a given collateral asset.
+    function getTotalStake(
+        address collateral
+    ) public view returns (uint256 amount) {
+        // Loop over all of the operators, get their middleware, and retrieve their staked amount.
+        for (uint256 i = 0; i < operators.length(); ++i) {
+            (address operator, IBoltManager.Operator memory operatorData) = operators.at(i);
+            amount += IBoltMiddleware(operatorData.middleware).getOperatorStake(operator, collateral);
+        }
+
+        return amount;
+    }
+
+    // ========= OPERATOR FUNCTIONS ====== //
+
+    /// @notice Registers an operator with Bolt. Only callable by a supported middleware contract.
+    function registerOperator(address operatorAddr, string calldata rpc) external onlyMiddleware {
+        if (operators.contains(operatorAddr)) {
+            revert OperatorAlreadyRegistered();
+        }
+
+        // Create an already enabled operator
+        Operator memory operator = Operator(rpc, msg.sender, Time.timestamp());
+
+        operators.set(operatorAddr, operator);
+    }
+
+    /// @notice De-registers an operator from Bolt. Only callable by a supported middleware contract.
+    function deregisterOperator(
+        address operator
+    ) public onlyMiddleware {
+        operators.remove(operator);
+    }
+
+    /// @notice Allow an operator to signal indefinite opt-out from Bolt Protocol.
+    /// @dev Pausing activity does not prevent the operator from being slashable for
+    /// the current network epoch until the end of the slashing window.
+    function pauseOperator(
+        address operator
+    ) external onlyMiddleware {
+        if (!operators.contains(operator)) {
+            revert OperatorNotRegistered();
+        }
```

Comment:
I don't think this is a necessary step since `disable` already reverts on a missing key. See https://github.com/chainbound/bolt/blob/jonas%2Ffeat%2Fregistry-updates/bolt-contracts/src/lib/EnumerableMap.sol#L49-L55. 
I get it is more explicit, it is just a gas cost consideration. You can ignore this if you think it's not relevant.